assemblies/assembly_architect-decision-services/states-con.adoc

```diff
@@ -0,0 +1,345 @@
+
+[source]
+----
+rule "Raise the alarm when we have one or more fires"
+when
+    exists Fire()
+then
+    insert( new Alarm() );
+    System.out.println( "Raise the alarm" );
+end
+----
+
+Likewise, when there are no fires we want to remove the alarm, so the `not` keyword can be used again.
+
+[source]
+----
+rule "Cancel the alarm when all the fires have gone"
+when
+    not Fire()
+    $alarm : Alarm()
+then
+    delete( $alarm );
+    System.out.println( "Cancel the alarm" );
+end
+----
+
+Finally there is a general health status message that is printed when the application first starts and after the alarm is removed and all sprinklers are turned off.
+
+[source]
+----
+rule "Status output when things are ok"
+when
+    not Alarm()
+    not Sprinkler( on == true )
+then
+    System.out.println( "Everything is ok" );
+end
+----
+
+As we did in the stateless session example, the above rules should be placed in a single DRL file and saved into the resources folder of your Maven project or one of its subfolders. We can then obtain a `KieSession` from the ``KieContainer``.
+The only difference is that this time we create a stateful session, whereas before we created a stateless session.
+
+[source,java]
+----
+KieServices kieServices = KieServices.Factory.get();
+KieContainer kContainer = kieServices.getKieClasspathContainer();
+KieSession ksession = kContainer.newKieSession();
+----
+
+
+With the session created you can now iterate your work with it. Four `Room` objects are created and inserted, as well as one `Sprinkler` object for each room. At this point the engine has done all of its matching, but no rules have fired yet.
+Calling `ksession.fireAllRules()` allows the matched rules to fire, but without a fire that will just produce the health message.
+
+[source,java]
+----
+String[] names = new String[]{"kitchen", "bedroom", "office", "livingroom"};
+Map<String,Room> name2room = new HashMap<String,Room>();
+for( String name: names ){
+    Room room = new Room( name );
+    name2room.put( name, room );
+    ksession.insert( room );
+    Sprinkler sprinkler = new Sprinkler( room );
+    ksession.insert( sprinkler );
+}
+
+ksession.fireAllRules();
+----
+
+[source]
+----
+> Everything is ok
+----
+
+Now, we will create two fires and insert them; this time a reference is kept for the returned ``FactHandle``. A fact handle is an internal engine reference to the inserted instance and allows instances to be retracted or modified at a later point in time.
+With the fires now in the engine, once `fireAllRules()` is called, the alarm is raised and the respective sprinklers are turned on.
+
+[source,java]
+----
+Fire kitchenFire = new Fire( name2room.get( "kitchen" ) );
+Fire officeFire = new Fire( name2room.get( "office" ) );
+
+FactHandle kitchenFireHandle = ksession.insert( kitchenFire );
+FactHandle officeFireHandle = ksession.insert( officeFire );
+
+ksession.fireAllRules();
+----
```

Comment:
Could we please update, same reasons as above:
```java
Fire kitchenFire = new Fire(name2room.get("kitchen"));
Fire officeFire = new Fire(name2room.get("office"));

FactHandle kitchenFireHandle = ksession.insert(kitchenFire);
FactHandle officeFireHandle = ksession.insert(officeFire);

ksession.fireAllRules();
```

Reply:
Done